include/grpc/passive_listener_injection.h

```diff
@@ -24,10 +24,14 @@
 namespace grpc_core {
 class ListenerInterface;
 class Server;
+namespace experimental {
+class PassiveListenerImpl;
+}  // namespace experimental
 }  // namespace grpc_core
 
-grpc_core::ListenerInterface* grpc_server_add_passive_listener(
-    grpc_core::Server* server, grpc_server_credentials* credentials);
+absl::Status grpc_server_add_passive_listener(
+    grpc_core::Server* server, grpc_server_credentials* credentials,
+    grpc_core::experimental::PassiveListenerImpl& passive_listener);
 
 // Called to add an endpoint to passive_listener.
 absl::Status grpc_server_accept_connected_endpoint(
```

Comment:
I think we no longer need this function.  Instead, `PassiveListenerImpl` can directly call the listener method.

As a result, I think we also no longer need the forward declaration of `ListenerInterface`, which in turn means that it can go back to being nested inside of `Server`.